src/main/java/io/r2dbc/proxy/callback/CallbackSupport.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.proxy.callback;
+
+import io.r2dbc.proxy.core.ConnectionInfo;
+import io.r2dbc.proxy.core.MethodExecutionInfo;
+import io.r2dbc.proxy.core.ProxyEventType;
+import io.r2dbc.proxy.core.QueryExecutionInfo;
+import io.r2dbc.proxy.listener.ProxyExecutionListener;
+import io.r2dbc.spi.Result;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Defines methods to augment execution of proxy methods used by child classes.
+ *
+ * @author Tadaya Tsuyukubo
+ */
+public abstract class CallbackSupport {
```

Comment:
How about naming types in this hierarchy `InvocationHandler` (this class would be named `InvocationHandlerSupport`)? Reflects better the actual purpose.

Reply:
I'm thinking to rename them to `CallbackHandler`.
So, 
- `CallbackSupport` => `CallbackHandlerSupport`
- `ReactiveStatementCallback` => `StatementCallbackHandler`
- `ReactiveConnectionFactoryCallback` => `ConnectionFactoryCallbackHandler`
- etc.